### ISSUE TITLE
Process creation has failed: Access denied.

### ISSUE THREAD

**Issue by Flitskikker** [lordmulder/MParallel#2]
Hi there,

While this tool looks really promising, I just can't get it to work on my pc (Windows 10 x64).
It's spawning this error for every command I feed into it, even the examples from the readme.

````
Process creation has failed: Access denied.

ERROR: Process ``ping.exe -n 16 gnu.org´´could not be created!

ping.exe -n 16 w3c.org
````

Things I've tried:
* Running cmd or MParallel as admin: Doesn't work.
* Running " in compatibility mode: Doesn't work.
* x86 vs x64 exe: Doesn't work.
* Older version: Doesn't work.

Any ideas? Thanks in advance!

**Comment by lordmulder** [comment]
Hello.

You should _not_ be running in "compatibility mode". It is evil! And running MParallel as "admin" is _only_ needed if whatever processes (tasks) your are creating need to run with "admin" rights.

Please post your complete and exact command-line! I'm on Windows 10 x64 as well. This works fine:

![image](https://user-images.githubusercontent.com/470590/30349771-61482df2-9814-11e7-8173-ac9092b0f258.png)

Regards
MuldeR

-----

BTW: If you have any Crapware (i.e. "anti-virus" software) installed on your computer, please un-install and then try again. More often than not, this kind of problem is caused by Crapware (i.e. "anti-virus" software).

**Comment by Flitskikker** [comment]
Thanks for your fast reply!

I tried the same command as yours, but I get the same result:

![image](https://user-images.githubusercontent.com/3516155/30350400-c07bf41e-9816-11e7-9e42-a21540a4666f.png)

Disabling ESET doesn't seem to have effect.

**Comment by lordmulder** [comment]
Well, so it looks like **CreateProcessW()**, which is a Windows system function, failed to launch "ping.exe" with Win32 error code `0x5`, aka "Access denied". There are like a zillion of reasons why the access to a file might be denied. But, if "ping.exe" starts normally when you run it *directly* (from the same console window), then something **weird** is going and I suspect that Crapware (i.e. "anti-virus" software) might be interfering.

To further analyze the problem, you might use the [**Process Monitor**](https://technet.microsoft.com/de-de/sysinternals/processmonitor.aspx). But, it will probably only show that MParallel called `CreateProcessW()` with the correct parameters and still `CreateProcessW()` return with error code `0x05`.

Anyway, did you try with some other program than "ping.exe"?

Regards
MuldeR

**Comment by lordmulder** [comment]
So, it looks like **`CreateProcessW()`**, which is a Windows system function, failed to launch "ping.exe" with error code `0x5` (*"Access denied"*). There are like a zillion of reasons why the access to a file might be denied. But, if "ping.exe" starts normally when you run it *directly*, from the same console window, then something **weird** is going on - in which case I suspect that Crapware (i.e. "anti-virus" software) might be interfering.

To further analyze the problem, you might use [**Process Monitor**](https://technet.microsoft.com/de-de/sysinternals/processmonitor.aspx). But, it will probably only show that MParallel called `CreateProcessW()` with the correct parameters, but it returned with error code `0x05`.

Under _normal_ circumstances, it should look like this:

![image](https://user-images.githubusercontent.com/470590/30351044-cadc90dc-9819-11e7-89d2-a2be168f5707.png)

**Anyway, did you try with some other program than "ping.exe"?**

Regards
MuldeR

**Comment by Flitskikker** [comment]
I checked with Process Monitor, but there weren't any Process Create calls.

I tried on my laptop, and there it worked (with the same "crapware").

I was really puzzled by this, so I just restarted my computer. Now it miraculously works! Even with ESET active.
Really strange...

Thanks for the tool! My tasks are now finished in a matter of seconds instead of a few minutes processing a single queue.
